### ISSUE TITLE
Recommender system benchmark

### ISSUE THREAD

**Issue by lkuper** [IntelLabs/ParallelAccelerator.jl#90]
From @ViralBShah :

> https://github.com/JuliaLang/julia/issues/15874
> 
> Folks, this is another non-trivial benchmark. It is the ALS algorithm. We have some work to improve the multi-threading performance, but figured that the folks at Intel may find it interesting. Abhijith, from Julia Computing, has been doing this work. The issue describes where the code is and how to run it with the datasets.
> 
> Good example for ParallelAccelerator too.

This might be interesting to look into for HPAT, too.

**Comment by ehsantn** [comment]
Yes, this is interesting for both ParallelAccelerator and HPAT. We have already looked at ALS to some extent for HPAT but we need sparse matrix support first.

**Comment by ViralBShah** [comment]
Yes, sparse matrix support is a must.

**Comment by lkuper** [comment]
For background on what the RecSys.jl package is doing, here's the [Julia Computing blog post on it](http://juliacomputing.com/blog/2016/04/22/a-parallel-recommendation-engine-in-julia.html).  And, to understand *that*, the context from the [ALS paper](http://www.grappa.univ-lille3.fr/~mary/cours/stats/centrale/reco/paper/MatrixFactorizationALS.pdf) is helpful.
